templates/directory.php

```diff
@@ -426,21 +427,23 @@ function pmpromd_shortcode($atts, $content=null, $code="")
 			{
 				foreach($theusers as $auser):
 					$auser = get_userdata($auser->ID);
+					var_dump($auser);
 					$auser->membership_level = pmpro_getMembershipLevelForUser($auser->ID);
+					$user_identifier = apply_filters( 'pmpromd_user_identifier', 'user_nicename' );
 					?>
 					<div id="pmpro_member-<?php echo $auser->ID; ?>" class="pmpro_member_directory-item">
 						<?php if(!empty($show_avatar)) { ?>
 							<div class="pmpro_member_directory_avatar">
 								<?php if(!empty($link) && !empty($profile_url)) { ?>
-									<a class="<?php echo $avatar_align; ?>" href="<?php echo add_query_arg('pu', $auser->user_nicename, $profile_url); ?>"><?php echo get_avatar($auser->ID, $avatar_size, NULL, $auser->display_name); ?></a>
+									<a class="<?php echo $avatar_align; ?>" href="<?php echo add_query_arg('pu', $auser->{$user_identifier}, $profile_url); ?>"><?php echo get_avatar($auser->ID, $avatar_size, NULL, $auser->display_name); ?></a>
```

Comment:
We should esc_url() the add_query_arg() here before echoing. We should also esc_attr() the avatar align var.